Download assets method

enriquegh opened the pull request:
Fixes #23

Commit:
add requirements.txt for dev testing

Commit:
add get asset content method

Commit:
add tests for job asset content method

cgoldberg reviewed (dismissed):
:+1:  thanks!

cgoldberg commented:
@enriquegh can you look at: https://travis-ci.org/cgoldberg/sauceclient/jobs/522345137 and fix in your branch?

Commit:
make default content_type an empty string instead of None
Prior to this change, the content_type when getting assets was set to None.
This didn't work on http.client module used by Python 3.

This change makes the content type header an empty string

Commit:
change open methods to use with statements

Commit:
force the creation of attribute

enriquegh commented:
@cgoldberg had to beat Python 3.4 to submission but the build finally passed 🎉
